python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+            two sets of flat with exposure times `flat_base_exptime, 2*flat_base_exptime, 4*flat_dn_range`,
+            `flat_dn_range = [1., 2., 4.]`. Default is [1., 2., 4., 8., 16., 32., 64.].
+        filter_id : `int`
+            Id of the filter to use in the ATSpectrograph. If `None` (default) will ignore the setup step.
+        grating_id : `int`
+            Id of the grating to use in the ATSpectrograph. If `None` (default) will ignore the setup step.
+        linear_stage : `float`
+            Position of the linear stage in the ATSpectrograph. If `None` (default) will ignore the
+            setup step.
+        read_out_time :`float`
+            The read out time of the camera, in seconds. Used to estimate script duration.
+
+        """
+
+        if n_dark is not None and n_dark > 0:
+            self.n_dark = n_dark
+        elif n_dark is not None:
+            raise RuntimeError(f"Number of dark frames must be larger than 0, got {n_dark}")
+
+        if t_dark is not None and 0. < t_dark <= self.maximum_exp_time:
+            self.t_dark = t_dark
+        elif n_dark is not None:
+            raise RuntimeError(f"Dark exptime must be in the range "
+                               f"(0.,{self.maximum_exp_time}], got {t_dark}")
+
+        if n_bias is not None and n_bias > 0:
+            self.n_bias = n_bias
+        elif n_bias is not None:
+            raise RuntimeError(f"Number of bias frames must be larger than 0, got {n_bias}")
+
+        if n_flat is not None and n_flat > 0:
+            self.n_flat = n_flat
+        elif n_flat is not None:
+            raise RuntimeError(f"Number of flat field frames must be larger than 0, got {n_flat}")
+
+        if flat_base_exptime is not None and flat_base_exptime > 0.:
+            self.flat_base_exptime = flat_base_exptime
+        elif flat_base_exptime is not None:
+            raise RuntimeError(f"Base flat field exposure time must be larger than 0., "
+                               f"got {flat_base_exptime}")
+
+        if flat_dn_range is not None:
+            self.flat_dn_range = np.array(flat_dn_range, dtype=float)
+
+        if np.any(self.flat_dn_range <= 0.):
+            n_bad = len(np.where(self.flat_dn_range <= 0.)[0])
+            raise RuntimeError(f"'flat_dn_range' must be larger than zero. "
+                               f"Got {n_bad} bad values.")
+
+        larger_flat_exptime = np.max(self.flat_dn_range)*self.flat_base_exptime
+        if larger_flat_exptime > self.maximum_exp_time:
+            raise RuntimeError(f"Flat field maximum exposure time {larger_flat_exptime}s "
+                               f"above maximum allowed {self.maximum_exp_time}s.")
+
+        if filter_id is not None:
+            self.filter = filter_id
+
+        if grating_id is not None:
+            self.grating = grating_id
+
+        if linear_stage is not None:
+            self.linear_stage = linear_stage
```

Comment:
Please consider removing the "if" statements. I think just setting the values should work (the default is None, and that's just fine for these).